libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/Announcement.kt

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2025 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
+ * Please see LICENSE files in the repository root for full details.
+ */
+
+package io.element.android.libraries.designsystem.components
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement.spacedBy
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import io.element.android.compound.theme.ElementTheme
+import io.element.android.compound.tokens.generated.CompoundIcons
+import io.element.android.libraries.designsystem.preview.ElementPreview
+import io.element.android.libraries.designsystem.preview.PreviewsDayNight
+import io.element.android.libraries.designsystem.theme.components.Button
+import io.element.android.libraries.designsystem.theme.components.ButtonSize
+import io.element.android.libraries.designsystem.theme.components.Icon
+import io.element.android.libraries.designsystem.theme.components.Surface
+import io.element.android.libraries.designsystem.theme.components.Text
+import io.element.android.libraries.ui.strings.CommonStrings
+
+@Composable
+fun Announcement(
+    title: String,
+    description: String?,
+    type: AnnouncementType,
+    modifier: Modifier = Modifier,
+) {
+    when (type) {
+        is AnnouncementType.Informative -> InformativeAnnouncement(
+            title = title,
+            description = description,
+            isError = type.isError,
+            modifier = modifier,
+        )
+        is AnnouncementType.Actionable -> ActionableAnnouncement(
+            title = title,
+            description = description,
+            actionText = type.actionText,
+            onActionClick = type.onActionClick,
+            onDismissClick = type.onDismissClick,
+            modifier = modifier,
+        )
+    }
+}
+
+@Immutable
+sealed interface AnnouncementType {
+    data class Informative(val isError: Boolean = false) : AnnouncementType
```

Comment:
Small note about the parameter `isError`, maybe rename to `isCritical` to match the Compound semantic?